### PR TITLE
feat: PubSub improvement: ISubscriber unsub [MTT-2765]

### PR DESCRIPTION
**Pull request by pdeschain**
### Description

Turned out to be a one-liner change that would allow us to also Usubscribe from ISubscribers. 

### Issue Number(s)

MTT-2765
### Contribution checklist
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] JIRA ticket ID is in the PR title or at least one commit message
 - [x] Include the ticket ID number within the body message of the PR to create a hyperlink
 - [x] Added to changelog

### CODE REVIEW

**Comment by SamuelBellomo** [comment]
With this, I would have expected to see changes in callsites of ISubscriber no? What kinds of improvements can we do with this?

**Comment by SamuelBellomo** [comment]
Keep Dispose for bundled dispose, for single dispose, can use symmetrical unsubscribe.

**Comment by SamuelBellomo** [comment]
Adding this for all PRs that have been opened since this new flow has been added. Please add your changelog here [project changelog](../CHANGELOG.md) file and/or [package changelog](../Packages/com.unity.multiplayer.samples.coop/CHANGELOG.md) file

File: Assets/Scripts/Gameplay/UI/IPConnectionWindow.cs [hunk 1]
```diff
@@ -34,7 +35,7 @@ void Awake()
 
         void OnDestroy()
         {
-            m_Subscription.Dispose();
+            m_ConnectStatusSubscriber.Unsubscribe(OnConnectStatusMessage);
```
Comment (fernando-cortez) [inline Assets/Scripts/Gameplay/UI/IPConnectionWindow.cs:38]:
```suggestion
            m_ConnectStatusSubscriber?.Unsubscribe(OnConnectStatusMessage);
```
Right?

File: Assets/Scripts/Gameplay/UI/IPUIMediator.cs [hunk 2]
```diff
@@ -61,7 +62,7 @@ void Start()
 
         void OnDestroy()
         {
-            m_Subscription.Dispose();
+            m_ConnectStatusSubscriber.Unsubscribe(OnConnectStatusMessage);
```
Comment (fernando-cortez) [inline Assets/Scripts/Gameplay/UI/IPUIMediator.cs:65]: Same here too.

File: Assets/Scripts/Gameplay/UI/Lobby/LobbyJoiningUI.cs [hunk 3]
```diff
@@ -40,7 +41,7 @@ void OnDisable()
 
         void OnDestroy()
         {
-            m_Subscriptions?.Dispose();
+            m_LocalLobbiesRefreshedSub.Unsubscribe(UpdateUI);
```
Comment (fernando-cortez) [inline Assets/Scripts/Gameplay/UI/Lobby/LobbyJoiningUI.cs:44]: +1.

File: Assets/Scripts/Gameplay/UI/Lobby/LobbyUIMediator.cs [hunk 4]
```diff
@@ -65,7 +65,7 @@ void OnConnectStatus(ConnectStatus status)
 
         void OnDestroy()
         {
-            m_Subscriptions?.Dispose();
+            m_ConnectStatusSubscriber.Unsubscribe(OnConnectStatus);
```
Comment (fernando-cortez) [inline Assets/Scripts/Gameplay/UI/Lobby/LobbyUIMediator.cs:68]: +1.

File: Assets/Scripts/Gameplay/UI/UnityServicesUIHandler.cs [hunk 5]
```diff
@@ -86,7 +87,7 @@ void HandleLobbyError(UnityServiceErrorMessage error)
 
         void OnDestroy()
         {
-            m_Subscriptions?.Dispose();
+            m_ServiceErrorSubscription.Unsubscribe(ServiceErrorHandler);
```
Comment (fernando-cortez) [inline Assets/Scripts/Gameplay/UI/UnityServicesUIHandler.cs:90]: +1.

**Review by fernando-cortez — DISMISSED** [review]

**Review by SamuelBellomo — DISMISSED** [review]

**Review by SamuelBellomo — DISMISSED** [review]

**Review by LPLafontaineB — DISMISSED** [review]

File: CHANGELOG.md [hunk 6]
```diff
@@ -19,7 +19,7 @@ Additional documentation and release notes are available at [Multiplayer Documen
 * NetworkObjectSpawner handles dynamically spawning in-scene placed NetworkObjects (#717) - You can't place a NetworkObject in scene directly and destroy it at runtime. This PR showcases proper handling of NetworkObjects that you'd wish to place inside of scenes, but would still want to destroy at game-time. Examples of these are: Imps, VandalImps, ImpBoss. NetworkObjects such as doors, crystals, door switch, etc. remain the same, statically-placed in scene.
 * Quality levels settings set up for Desktop [MTT-4450] (#713)
 * Added custom RNSM config with graph for RTT instead of single value (#747)
-
+* Added Unsubscribe API for the ISubscriber<T> along with refactoring of the codebase to use this API instead of IDisposable handle when there is just one subscription.
```
Comment (fernando-cortez) [inline CHANGELOG.md:22]: Just nit to include PR #.

**Review by fernando-cortez — APPROVED** [review]

**Review by LPLafontaineB — APPROVED** [review]

**Pull request merged**